sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1666,10 +1679,12 @@ class Analyzer(
           // we need to check this and prevent applying this rule multiple times
           if (sortOrder == finalSortOrders) {
             sort
+          } else if (needsPushDown.isEmpty) {
+            Sort(finalSortOrders, global, child)
           } else {
-            Project(aggregate.output,
+            Project(child.output,
               Sort(finalSortOrders, global,
-                aggregate.copy(aggregateExpressions = originalAggExprs ++ needsPushDown)))
+                pushDownMissingAttrs(needsPushDown, child)))
```

Comment:
Seems the principle we followed now is to return original plan if we can't push down it. Should we also follow it here?

Reply:
not sure I got what you mean here, may you please explain me a bit more? Thanks.

Reply:
oh, for now `pushDownMissingAttrs` will return `child` if it is unexpected. For that case, we may just return original plan, instead of adding `Project` and changing `Sort`.

Reply:
I see. Well, actually in that case we'd have an invalid plan, since we'd be referencing a non-existing attribute. So I don't think we have to consider it, we'd fail anyway.

Reply:
In that case, it is better to keep original plan for investigating, I think. You can see that if we can't resolve aggregate expressions, we do just return original plan. By changing original plan like that, it makes debugging harder.

Reply:
in that case we can maybe just throw the AnalysisException in the other case so the original sort is kept. Let me do that then, thanks.